src/main/java/eu/ha3/presencefootsteps/world/LocomotionLookup.java

```diff
@@ -1,26 +1,27 @@
 package eu.ha3.presencefootsteps.world;
 
-import java.util.LinkedHashMap;
-import java.util.Map;
-
 import eu.ha3.presencefootsteps.PresenceFootsteps;
 import eu.ha3.presencefootsteps.sound.generator.Locomotion;
+import it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityType;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.registry.Registries;
 import net.minecraft.util.Identifier;
 
-public class LocomotionLookup implements Index<Entity, Locomotion> {
+import java.util.Map;
+
+public record LocomotionLookup(Map<Identifier, Locomotion> values) implements Index<Entity, Locomotion> {
 
-    private final Map<Identifier, Locomotion> values = new LinkedHashMap<>();
+    public LocomotionLookup() {
+        this(new Object2ObjectLinkedOpenHashMap<>());
+    }
 
     @Override
     public Locomotion lookup(Entity key) {
-        if (key instanceof PlayerEntity) {
-            return Locomotion.forPlayer((PlayerEntity)key, Locomotion.NONE);
-        }
-        return Locomotion.forLiving(key, values.getOrDefault(EntityType.getId(key.getType()), Locomotion.BIPED));
+        return key instanceof PlayerEntity player
+                ? Locomotion.forPlayer(player, Locomotion.NONE)
+                : Locomotion.forLiving(key, values.getOrDefault(EntityType.getId(key.getType()), Locomotion.BIPED));
```

Comment:
Revert this please for readability.

Reply:
Is this really that bad? It looks very comprehensible to me: the ternary operator is at the method return and shouldn't confuse that much.

Reply:
Maybe for you, but for me I'd rather have a separate if statement for this kind of thing. I also want to ensure the PR is lean before I merge as it makes it easier for me to know what's affected.

Reply:
Fine, I'm going to revert this.